src/main/java/com/rackspace/salus/common/web/AbstractRestExceptionHandler.java

```diff
@@ -179,4 +181,14 @@ protected void logRequestFailure(HttpServletRequest request, Exception e) {
     return respondWith(request, HttpStatus.SERVICE_UNAVAILABLE, ResponseMessages.kafkaExceptionMessage);
   }
 
+  @ExceptionHandler(MethodArgumentNotValidException.class)
+  public ResponseEntity<?> handleMethodArgumentNotValidException(HttpServletRequest request, MethodArgumentNotValidException e) {
+    logRequestFailure(request, e);
+    String message = "One or more field validations failed: " +
+        e.getBindingResult().getFieldErrors().stream()
+            .map(error -> error.getField() + " " + error.getDefaultMessage())
```

Comment:
@jjbuchan @GeorgeJahad , what were your thoughts about whether or not to emphasize the field name in the resulting message? Single quotes around the field name might be a subtle but helpful way to call that out.

Reply:
I think I prefer keeping this message simpler.  If it was only representing a single failure it would be good, but since it is all failures including the field and the message could become massive.

I'm leaning towards only specifying the field names in this and then faraji can pull out the `message`s from the `errors` list.  I feel like that would display the errors more cleanly to the customer than one massive string.

Reply:
I think this is the change that would work
```suggestion
            .map(error -> error.getField())
```

Reply:
Does that mean I should remove the changes in PR:-
https://github.com/racker/salus-telemetry-etcd-adapter/pull/68/files
?

Reply:
That change is still good

Reply:
Yeah, we'd still expect that new error to turn up in `errors`, right?

Reply:
Yes...`errors` is an array object, So it will have multiple errors coming.